controllers/process.go

```diff
@@ -214,22 +242,10 @@ func (r *ServicefenceReconciler) ReconcileNamespace(req ctrl.Request) (ret ctrl.
 
 // refreshFenceStatusOfService caller should hold the reconcile lock.
 func (r *ServicefenceReconciler) refreshFenceStatusOfService(ctx context.Context, svc *corev1.Service, nsName types.NamespacedName) (reconcile.Result, error) {
-	if svc == nil {
-		// Fetch the Service instance
-		svc = &corev1.Service{}
-		err := r.Client.Get(ctx, nsName, svc)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				svc = nil
-			} else {
-				log.Errorf("get service %s error, %+v", nsName, err)
-				return reconcile.Result{}, err
-			}
-		}
-	} else {
+	if svc != nil {
```

Comment:
how will it goes when svc == nil?

Reply:
it will use nsName directly, fence will be deleted actually if exists